internal/security/setup/option/optionHelpers.go

```diff
@@ -214,20 +233,64 @@ func markComplete(dirPath string) error {
 	return nil
 }
 
-func getXdgRuntimeDir() string {
+func getWorkDir() string {
+	var workDir string
+	var err error
+
 	if xdgRuntimeDir, ok := os.LookupEnv(envXdgRuntimeDir); ok {
-		return xdgRuntimeDir
+		workDir = filepath.Join(xdgRuntimeDir, pkiInitBaseDir)
+	} else if setup.Configuration.SecretsSetup.WorkDir != "" {
+		workDir, err = filepath.Abs(setup.Configuration.SecretsSetup.WorkDir)
+		if err != nil {
+			// use default work dir if error occurs
+			workDir = filepath.Join(defaultWorkDir, pkiInitBaseDir)
+		}
+	} else {
+		// if not specified in config and $XDG_RUNTIME_DIR is undefined, default to /tmp
+		workDir = filepath.Join(defaultWorkDir, pkiInitBaseDir)
+	}
+
+	return workDir
+}
+
+func getCertConfigDir() (string, error) {
+	var certConfigDir string
+	if setup.Configuration.SecretsSetup.CertConfigDir != "" {
+		certConfigDir = setup.Configuration.SecretsSetup.CertConfigDir
+		if exist, _ := checkDirExistsAndIsWritable(certConfigDir); !exist {
+			return "", fmt.Errorf("CertConfigDir from config file does not exist in: %s", certConfigDir)
+		}
+
+		return certConfigDir, nil
 	}
-	// if $XDG_RUNTIME_DIR is undefined, default to /tmp
-	return defaultXdgRuntimeDir
+
+	return "", errors.New("Directory for certificate configuration files not configured")
 }
 
-func getPkiCacheDirEnv() string {
-	pkiCacheDir := os.Getenv(envPkiCache)
-	if pkiCacheDir == "" {
-		return defaultPkiCacheDir
+func getCacheDir() (string, error) {
+	cacheDir := defaultPkiCacheDir
+
+	if setup.Configuration.SecretsSetup.CacheDir != "" {
+		cacheDir = setup.Configuration.SecretsSetup.CacheDir
+		if exist, isWritable := checkDirExistsAndIsWritable(cacheDir); !exist || !isWritable {
```

Comment:
I'd prefer we return separate error conditions for this, i.e.

```suggestion
		exist, isWritable := checkDirExistsAndIsWritable(cacheDir)
		if !exist {
		     return "", fmt.Errorf("CacheDir from config file, %s, does not exist", cacheDir)
		 }
		 if !isWritable {
		     return "", fmt.Errorf("CacheDir from config file, %s, is not writable", cacheDir)
		 }
```